crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -446,32 +446,19 @@ impl<M: NetworkMsg, K: SignatureKey + 'static> Libp2pNetwork<M, K> {
                 };
                 handle.add_known_peers(bs_addrs).await.unwrap();
 
-                // 10 minute timeout
-                let timeout_duration = Duration::from_secs(600);
-                // perform connection
-                info!("WAITING TO CONNECT ON NODE {:?}", id);
-                handle
-                    .wait_to_connect(4, id, timeout_duration)
-                    .await
-                    .unwrap();
-
-                let connected_num = handle.num_connected().await?;
-                metrics_connected_peers
-                    .metrics
-                    .connected_peers
-                    .set(connected_num);
+                handle.begin_bootstrap().await?;
```

Comment:
So one interesting thing is with this change, the wait for connect function would actually never return.  I'm actually not sure what is going on there but it seems like constantly looking up peers was causing connections to happen.  My thought was that if the put queries below this could succeed then we must have the required connectivity for normal operation and should have the same effect.  If I understand correctly, the put query succeeding would mean we were able to reach a quorum of nodes (replication factor is 2f + 1) guarantee the same thing.  

I think we should have periodic logging/metrics for how many peers we are connected to.

Reply:
I'm really unsure what is going on, it's like triggering the bootstrap alone wasn't starting the gossip layer or something.  When I just let it run, it's clear that gossip is connecting to peers and just fine as our broadcast succeed normally.

Reply:
For now I'm going to add this back in, but after we do all the put queries and subscribe to topics.  It seems at that point we have all the connections we need.